src/test/java/com/thirdcc/webapp/web/rest/DebtResourceIT.java

```diff
@@ -98,188 +115,135 @@ public void setup() {
-        restDebtMockMvc.perform(post("/api/debts")
-            .contentType(TestUtil.APPLICATION_JSON_UTF8)
-            .content(TestUtil.convertObjectToJsonBytes(debtDTO)))
-            .andExpect(status().isBadRequest());
-
-        // Validate the Debt in the database
-        List<Debt> debtList = debtRepository.findAll();
-        assertThat(debtList).hasSize(databaseSizeBeforeCreate);
+    
+    private void initDebtDB() {
+        debtRepository.saveAndFlush(debt);
+    }
+    
+    private void initEventDB() {
+        event = createEventEntity();
+        eventRepository.saveAndFlush(event);
     }
-
 
     @Test
-    @Transactional
-    public void getAllDebts() throws Exception {
+    public void getAllDebtsByEventId() throws Exception {
+        //TODO implement checking for admin and event crew
         // Initialize the database
-        debtRepository.saveAndFlush(debt);
+        initDebtDB();
+        initEventDB();
 
         // Get all the debtList
-        restDebtMockMvc.perform(get("/api/debts?sort=id,desc"))
+        restDebtMockMvc.perform(get("/api/debts/event/{eventId}?sort=id,desc", event.getId()))
             .andExpect(status().isOk())
             .andExpect(content().contentType(MediaType.APPLICATION_JSON_UTF8_VALUE))
             .andExpect(jsonPath("$.[*].id").value(hasItem(debt.getId().intValue())))
             .andExpect(jsonPath("$.[*].receiptId").value(hasItem(DEFAULT_RECEIPT_ID.intValue())))
             .andExpect(jsonPath("$.[*].eventAttendeeId").value(hasItem(DEFAULT_EVENT_ATTENDEE_ID.intValue())))
-            .andExpect(jsonPath("$.[*].amount").value(hasItem(DEFAULT_AMOUNT.intValue())))
+            .andExpect(jsonPath("$.[*].amount").value(hasItem(DEFAULT_AMOUNT.doubleValue())))
             .andExpect(jsonPath("$.[*].status").value(hasItem(DEFAULT_STATUS.toString())));
     }
     
     @Test
     @Transactional
-    public void getDebt() throws Exception {
-        // Initialize the database
-        debtRepository.saveAndFlush(debt);
-
-        // Get the debt
-        restDebtMockMvc.perform(get("/api/debts/{id}", debt.getId()))
-            .andExpect(status().isOk())
-            .andExpect(content().contentType(MediaType.APPLICATION_JSON_UTF8_VALUE))
-            .andExpect(jsonPath("$.id").value(debt.getId().intValue()))
-            .andExpect(jsonPath("$.receiptId").value(DEFAULT_RECEIPT_ID.intValue()))
-            .andExpect(jsonPath("$.eventAttendeeId").value(DEFAULT_EVENT_ATTENDEE_ID.intValue()))
-            .andExpect(jsonPath("$.amount").value(DEFAULT_AMOUNT.intValue()))
-            .andExpect(jsonPath("$.status").value(DEFAULT_STATUS.toString()));
-    }
-
-    @Test
-    @Transactional
-    public void getNonExistingDebt() throws Exception {
+    public void getDebt_NonExisting_ShouldReturn404() throws Exception {
+        initDebtDB();
+        
         // Get the debt
         restDebtMockMvc.perform(get("/api/debts/{id}", Long.MAX_VALUE))
```

Comment:
the REST API does not exist it must throw 404
Ii not sure is it a suitable test

Reply:
i am thinking that the checking is inside the method so should make a test case for it as well